lib/httpd_configmap_generator/ldap.rb

```diff
@@ -0,0 +1,190 @@
+module HttpdConfigmapGenerator
+  class Ldap < Base
+    AUTHCONFIG_COMMAND = "/usr/sbin/authconfig".freeze
+    LDAP_MODES = %w(ldap ldaps).freeze
+
+    AUTH = {
+      :type    => "external",
+      :subtype => "ldap"
+    }.freeze
+
+    def required_options
+      super.merge(
+        :cert_file   => { :description => "Cert File" },
+        :ldap_host   => { :description => "LDAP Directory Host FQDN" },
+        :ldap_mode   => { :description => "ldap | ldaps" },
+        :ldap_basedn => { :description => "LDAP Directory Base DN" },
+      )
+    end
+
+    def optional_options
+      super.merge(
+        :ldap_group_name         => { :description => "LDAP Directory Group Name",
+                                      :default     => "cn" },
+        :ldap_group_member       => { :description => "Attribute containing the names of the group's member",
+                                      :default     => "member" },
+        :ldap_group_object_class => { :description => "The object class of a group entry in LDAP",
+                                      :default     => "groupOfNames" },
+        :ldap_id_use_start_tls   => { :description => "Must connection also use tls",
+                                      :default     => "true" },
```

Comment:
booleans must be :default => true  otherwise Trollop thinks it's a string if "true"